File: 1º Ano/2º Semestre/Programação Orientada a Objetos/Labs/Lab06/CardGames/src/CardGames/SuecaDeck.java
package CardGames;

/**
 * Representa um baralho de cartas usado no jogo da sueca.
 * 
 * @author POO 2019/2020
 * @version abr/2020
 */
// Nível 5
public class SuecaDeck extends Deck {

    private final static int ACE_VALUE = 11;
    private final static int SEVEN_VALUE = 10;
    private final static int QUEEN_VALUE = 2;
    private final static int JACK_VALUE = 3;
    private final static int KING_VALUE = 4;

    public SuecaDeck() {

        createCards();
    }

    private void createCards() {
        clear();
        for (Suit suit : Suit.values()) {
            if (suit == Suit.NONE) {
                continue;
            }
            addCard(new NumberedCard(1, suit, ACE_VALUE));
            for (int i = 2; i <= 6; i++) {
                addCard(new NumberedCard(i, suit, 0));
            }
            addCard(new NumberedCard(7, suit, SEVEN_VALUE));
            addCard(new FaceCard(FaceName.QUEEN, suit, QUEEN_VALUE));
            addCard(new FaceCard(FaceName.JACK, suit, JACK_VALUE));
            addCard(new FaceCard(FaceName.KING, suit, KING_VALUE));
        }
    }
}

File: 3º Ano/Anual/Programação Visual (PV)/Labs/Lab 1/Lab1/SuecaDeck.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab1
{
    class SuecaDeck : Deck
    {
        private static int ACE_VALUE = 11;
        private static int SEVEN_VALUE = 10;
        private static int QUEEN_VALUE = 2;
        private static int JACK_VALUE = 3;
        private static int KING_VALUE = 4;

        public SuecaDeck()
        {
            createCards();
        }
        private void createCards()
        {
            base.Clear();
            foreach (Suit suit in Enum.GetValues(typeof(Suit)))
            {
                if (suit == Suit.None)
                {
                    continue;
                }
                base.AddCard(new NumberedCard(1, suit, ACE_VALUE));
                for (int i = 2; i <= 6; i++)
                {
                    base.AddCard(new NumberedCard(i, suit, 0));
                }
                base.AddCard(new NumberedCard(7, suit, SEVEN_VALUE));
                base.AddCard(new FaceCard(FaceName.Queen, suit, QUEEN_VALUE));
                base.AddCard(new FaceCard(FaceName.Jack, suit, JACK_VALUE));
                base.AddCard(new FaceCard(FaceName.King, suit, KING_VALUE));
            }
        }

    }
}
